Assets/Scripts/GameProcess/HandDrower.cs

```diff
@@ -17,15 +19,16 @@ public static Vector3 calculateStartHandPosition(Vector3 handPositionCoordinate)
     }
 
 
-    public void fillStartHand(GameObject cardParent, Vector3 cardPosition, Main _mainScript)
+    public void fillStartHand(GameObject cardParent, Vector3 cardPosition, List<Card> cards, Dictionary<Guid, GameObject> cardsInScene, Action<SimpleCard> clickAction)
     {
         int offset = 2;
 
-        for (int card_counter = 0; card_counter < HandDrower.handLimit; card_counter++)
+        for (int card_counter = 0; card_counter < cards.Count; card_counter++)
```

Comment:
Not sure why this condition has been changed. The logic line behind it is that hand is being filled by particular number of cards when game started. This number doesn't depend on cards count (assume that cards count is always the same for both players) but just on static value that is pre-defined (in that case it's 5). I also think that we need 2 variables here: one will describe max number of cards in hand that is allowed and another one (that should be used in that case) is initial number of cards in player's hand

Reply:
The reason I've changed it is because before cards were created without connection to what player has selected in CardSelection scene. After I started using actual selection, I switched to cards.Count, since it represents number of cards in player's selection. I think using HandDrower.handLimit is not correct, since number of cards passed to the method can possibly be lower.

Reply:
> ... since number of cards passed to the method can possibly be lower.
Agree. But initial number of cards should be specified. It doesn't matter how many cards in selection are (since this number will be always the same). Let's say for now that selection will contain more than 3 cards and player will get 3 cards at the beginning? 

For that purpose I'd like to add another variable that will be used for that (`initial_card_count` , etc.)

Reply:
I've added initialDrawSize